crates/subspace-farmer/src/single_disk_farm/unbuffered_io_file_windows.rs

```diff
@@ -103,9 +109,17 @@ impl UnbufferedIoFileWindows {
         open_options.advise_unbuffered();
         let read_file = open_options.read(true).open(path)?;
 
+        // Physical sector size on many SSDs is smaller than 4096 and should improve performance
+        let physical_sector_size = if read_file.read_at(&mut [0; 512], 0).is_ok() {
```

Comment:
Is this the proper way to determine if the sector size is smaller than 4096? from the [doc](https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html#tymethod.read_at) of `read_at` it didn't mention in which case it will fail

Reply:
As mentioned in the PR description this is a good enough way (from my point of view) to do that.

> from the doc of read_at it didn't mention in which case it will fail

First of all this is a different `read_at`, second the actual requirement comes from `FILE_FLAG_NO_BUFFERING` we are using (see implementation of `advise_unbuffered`): https://learn.microsoft.com/en-us/windows/win32/fileio/file-buffering#alignment-and-file-access-requirements